### PR TITLE
Update docs for v1.2

### PR DESCRIPTION
**Pull request by FxKu**
- Make README more compact and move parts to index.md
- Update quickstart with kind, OLM
- Remove paragraphs from dev guide which are redundant to quickstart
- Move user-facing clusterrole section to admin docs
- Add standby section and mark optional parameters in cluster_manifest.md
- Have a heading and intro sentence in each guide
- Fix some heading hierarchies and try to only have 80-character-long lines

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,40 +8,25 @@
 
 <img src="docs/diagrams/logo.png" width="200">
 
-## Introduction to the Postgres Operator
+The Postgres operator manages PostgreSQL clusters on Kubernetes. Configuration
+happens only via manifests. Therefore, it integrates easily in automated CI/CD
+pipelines with no access to Kubernetes directly.
 
-The Postgres [operator](https://coreos.com/blog/introducing-operators.html)
```
Comment (Jan-M) [inline README.md]: Where did this go?

File: README.md [hunk 2]
```diff
@@ -51,26 +36,16 @@ The Postgres Operator made it to the [Google Summer of Code 2019](https://summer
 * [cluster manifest reference](docs/reference/cluster_manifest.md)
 * [command-line options and environment variables](docs/reference/command_line_and_environment.md)
 
-The rest of this document is a tutorial to get you up and running locally with the operator on Minikube.
-
-## Overview of involved entities
```
Comment (Jan-M) [inline README.md]: Where are the entities now?

File: README.md (hunk 1 above)
Comment (FxKu) [reply README.md]: "Moved" to index.md (concepts). It already contained the same description as the README except for the 4th bullet point.

File: README.md (hunk 2 above)
Comment (FxKu) [reply README.md]: Moved to index.md as well. I thought it's a better place than the README, which is quite bloated right now.

**Comment by sdudoladov** [comment]
:+1:

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Comment by RafiaSabih** [comment]
:+1:

**Pull request merged**